server/app/models/grid_service.rb

```diff
@@ -67,8 +67,12 @@ def to_path
   end
 
   # @return [String]
-  def name_with_stack
-    "#{self.stack.name}-#{self.name}"
+  def name_with_stack(ignore_default_stack: false, stack_separator: '-')
```

Comment:
* Without the `stack_separator: '-'`; a "path" should always be `/`-separated?
* The `ignore_foo: false` double-negation sounds odd

This is some kind of "path relative to grid" used within the grid vs the "absolute grid/stack/service" path used in the API URLs.... dunno what the correct name for it is.